src/PrestaShopBundle/Controller/Admin/Sell/Catalog/Product/CombinationController.php

```diff
@@ -56,15 +88,46 @@ public function getListAction(int $productId, Request $request): JsonResponse
             $offset ?? null
         ));
 
-        return $this->json($this->formatResponse($combinationsList));
+        return $this->json($this->formatListResponse($combinationsList));
+    }
+
+    /**
+     * @AdminSecurity("is_granted('update', request.get('_legacy_controller'))")
+     *
+     * @param int $combinationId
+     * @param Request $request
+     *
+     * @return JsonResponse
+     */
+    public function updateCombinationFromListingAction(int $combinationId, Request $request): JsonResponse
+    {
+        $form = $this->getCombinationItemFormBuilder()->getFormFor($combinationId, [], [
+            'method' => Request::METHOD_PATCH,
+        ]);
+        $form->handleRequest($request);
+
+        try {
+            $result = $this->getCombinationItemFormHandler()->handleFor($combinationId, $form);
+
+            if (!$result->isValid()) {
+                return $this->json(['message' => (string) $form->getErrors(true)], Response::HTTP_BAD_REQUEST);
```

Comment:
```suggestion
                return $this->json(['errors' => $this->getFormErrorsForJS($form)], Response::HTTP_BAD_REQUEST);
```

You'd better use the `getFormErrorsForJS` and adapt the JS to display an array, or even just the first one But the return is more adapted and it's translated (Well supposed to be anyway because I only got english messages while trying it 😅 ) But at least you get to know which field is invalid

Reply:
It is translated in form. And I've updated constraints in form so they mention field name which failed.
Btw we are submitting one field in one request, so I doubt there would be many errors. But ok i can do that, so your suggestion might bring some unnecessary complexity, like handling array of errors in js instead of simple "message" every failure :thinking: You sure its worth it?

Reply:
done